src/Features/Core/Portable/CodeRefactorings/ConvertToInterpolatedString/AbstractConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,260 @@
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var arguments = GetArguments(invocation).Value;
+            var literalExpression = syntaxFactsService.GetExpressionOfArgument(arguments[0]) as TLiteralExpressionSyntax;
+            string text = GetText(literalExpression);
+            var syntaxGenerator = document.Project.LanguageServices.GetService<SyntaxGenerator>();
+            var expandedArguments = GetExpandedArguments(semanticModel, arguments, syntaxGenerator, syntaxFactsService);
+            var interpolatedString = GetInterpolatedString(text);
+            var newInterpolatedString = VisitArguments(expandedArguments, interpolatedString, syntaxFactsService);
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var newRoot = root.ReplaceNode(invocation, newInterpolatedString.WithTriviaFrom(invocation));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private ImmutableArray<TExpressionSyntax> GetExpandedArguments(
+            SemanticModel semanticModel,
+            SeparatedSyntaxList<TArgumentSyntax> arguments,
+            SyntaxGenerator syntaxGenerator,
+            ISyntaxFactsService syntaxFactsService)
+        {
+            var builder = ImmutableArray.CreateBuilder<TExpressionSyntax>();
+            for (int i = 1; i < arguments.Count; i++)
+            {
+                var argumentExpression = syntaxFactsService.GetExpressionOfArgument(arguments[i]);
+                var convertedType = semanticModel.GetTypeInfo(argumentExpression).ConvertedType;
+                if (convertedType == null)
+                {
+                    
+                    builder.Add(syntaxFactsService.Parenthesize(argumentExpression) as TExpressionSyntax);
+                }
+                else
+                {
+                    var castExpression = syntaxGenerator.CastExpression(convertedType, syntaxFactsService.Parenthesize(argumentExpression)).WithAdditionalAnnotations(Simplifier.Annotation);
+                    builder.Add(castExpression as TExpressionSyntax);
+                }
+
+            }
+
+            var expandedArguments = builder.ToImmutable();
+            return expandedArguments;
+        }
+
+        private TInterpolatedStringExpressionSyntax VisitArguments(
+            ImmutableArray<TExpressionSyntax> expandedArguments,
+            TInterpolatedStringExpressionSyntax interpolatedString,
+            ISyntaxFactsService syntaxFactsService)
+        {
+            return interpolatedString.ReplaceNodes(GetInterpolatedStringContents(interpolatedString), (oldNode, newNode) =>
+            {
+                var interpolationSyntaxNode = newNode as TInterpolationSyntax;
+                if (interpolationSyntaxNode == null)
```

Comment:
you could invert this, and only do the following code if the expression != null.  Then you only need one location in the code where you have "return newNode".